src/engine/meta/meta_blks_mgr.cpp

```diff
@@ -235,6 +237,36 @@ void MetaBlkMgr::scan_meta_blks() {
     if (self_recover) { set_self_recover(); }
 }
 
+void MetaBlkMgr::create_sobject(client_info_map_t& sub_info, meta_sub_type type , uint64_t meta_bid){
+    auto sm{HomeStoreBase::safe_instance()->sobject_mgr()};
+    auto metablk_name{"MetaBlk_" + type};
+    auto metablk_type{"MetaBlk"};
+    // first time creating child and add to MetaBlkMgr
+    if (sub_info[type].meta_bids.size() == 1) {
+        auto meta_blk_obj =
+            sm->create_object(metablk_type, metablk_name,
+                              std::bind(&MetaBlkMgr::get_status_metablk, this, std::placeholders::_1, metablk_name));
+        m_sobject->add_child(meta_blk_obj);
+    }
+    // create child for each type and add them to child list
+    auto meta_blk_obj{m_sobject->get_child(metablk_name)};
+    auto metablk_meta_bids_name{metablk_name + "_" + std::to_string(meta_bid)};
+    auto metablk_meta_bids_type{metablk_name};
+    auto metablk_meta_bids_obj{sm->create_object(
```

Comment:
I didn't get why we need this level of object and child, shouldn't it be done through `MetaBlkMgr::get_status_metablk`?

Reply:
I think one of main objectives of sobject is to make the hierarchical points of view. For this class for example, the meta block of each volume is a child of metablk VOLUME.

Reply:
Shouldn't `--type MetaBlk --name MetaBlk_Volume` (which you've already created the object at line:248) already served the purpose to get all the vol meta blks?

Reply:
All registered volumes have the same type `MetaBlk_Volume`